hyperspy/datasets/artificial_data.py

```diff
@@ -356,3 +363,231 @@ def get_atomic_resolution_tem_signal2d():
+    %s
+    %s
+    random_state: None or int
+        initialise state of the random number generator
+
+    Example
+    -------
+    >>> import hyperspy.datasets.artifical_data as ad
+    >>> s = ad.get_luminescence_map_nonuniform()
+    >>> s.plot()
+
+    With constant baseline
+
+    >>> s = ad.get_luminescence_map_nonuniform(uniform=True, add_baseline=True)
+    >>> s.plot()
+
+    Make the noise the same for multiple spectra, which can
+    be useful for testing fitting routines
+
+    >>> s1 = ad.get_luminescence_map_nonuniform(random_state=10)
+    >>> s2 = ad.get_luminescence_map_nonuniform(random_state=10)
+    >>> (s1.data == s2.data).all()
+    True
+
+    See also
+    --------
+    get_low_loss_eels_signal,
+    get_core_loss_eels_signal,
+    get_low_loss_eels_line_scan_signal,
+    get_core_loss_eels_line_scan_signal,
+    get_core_loss_eels_model,
+    get_atomic_resolution_tem_signal2d,
+    get_luminescence_spectrum_nonuniform
+    """
+    from hyperspy.signals import Signal1D
+    from hyperspy import components1d
+    from hyperspy.axes import FunctionalDataAxis,UniformDataAxis
+
+
+    #Initialisation of random number generator
+    random_state = check_random_state(random_state)
+
+    #Creating a uniform data axis, roughly similar to Horiba iHR320 with a 150 mm-1 grating
+    nm_axis = UniformDataAxis(
+        index_in_array=None,
+        name="Wavelength",
+        units="nm",
+        navigate=False,
+        size=1024,
+        scale=0.54,
+        offset=222.495,
+        is_binned=False,
+        )
+    #Spatial axes
+    spax_x = UniformDataAxis(index_in_array=None,
+        name="X",
+        units="um",
+        navigate=False,
+        size=10,
+        scale=2.1,
+        offset=0,
+        is_binned=False,
+    )
+
+    spax_y = UniformDataAxis(index_in_array=None,
+        name="Y",
+        units="um",
+        navigate=False,
+        size=10,
+        scale=2.1,
+        offset=0,
+        is_binned=False,
+    )
+
+    #Artificial luminescence peak
+    gaussian_peak = components1d.Gaussian(A=5000, centre=375, sigma=25)
+
+    #Creating data array
+    data = np.zeros((100,1024))
+    #c-style works too!
+    for i in range(100):
+        #Creating data array, possibly with noise and baseline
+        data[i] = gaussian_peak.function(nm_axis.axis)
+        if add_noise:
+            data[i] += (random_state.uniform(size=len(nm_axis.axis)) - 0.5)*1.4
+        if add_baseline:
+            data[i] += 350.
+
+    #Creating the signal with axis and data
+    data = data.reshape((10,10,1024))
+    sig = Signal1D(data,axes=[spax_y,spax_x,nm_axis])
+    #sig.metadata.General.title = 'Artificial Luminescence map'
+
+    #if not uniform, transformation into non-linear axis
+    if not uniform:
+        hc = 1239.84198 #nm/eV
+
+        #eV axis creation. Note how the slice is inverted to have low energy first
+        evax = FunctionalDataAxis(expression="a/x",
```

Comment:
Would it be simpler to use `convert_to_functional_data_axis`?

Reply:
It was. I did